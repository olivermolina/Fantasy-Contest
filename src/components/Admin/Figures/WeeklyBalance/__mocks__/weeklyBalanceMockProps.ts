export const weeklyBalanceMockData = [
  {
    agent: null,
    id: '1',
    username: 'jdoe',
    firstname: 'John',
    lastname: 'Doe',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
  {
    agent: null,
    id: '2',
    username: 'jwick',
    firstname: 'John',
    lastname: 'Wick',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
];
