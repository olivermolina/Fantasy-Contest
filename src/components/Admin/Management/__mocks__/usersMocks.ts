import { faker } from '@faker-js/faker';
import { UserType } from '@prisma/client';

export const usersMock = [
  {
    id: '1',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [
      {
        id: '1',
        userId: '1',
        code: 'cod1',
      },
      {
        id: '2',
        userId: '1',
        code: 'code2',
      },
    ],
    type: UserType.AGENT,
  },
  {
    id: '2',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '3',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '4',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
];
