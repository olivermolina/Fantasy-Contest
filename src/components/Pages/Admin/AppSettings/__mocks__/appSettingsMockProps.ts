export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
