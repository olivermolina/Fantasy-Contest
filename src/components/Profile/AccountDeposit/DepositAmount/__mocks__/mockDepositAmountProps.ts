export const mockDepositAmountProps = {
  setDepositAmount: () => {
    alert('Set Deposit amount');
  },
  handleNext: () => {
    alert('handleNext');
  },
  depositAmount: 10,
  isFirstDeposit: false,
  maxMatchDeposit: 50,
  depositAmountOptions: [10, 25, 50, 75, 100, 125],
  reloadBonusAmount: 0,
  reloadBonusType: 'FLAT',
  showBonusCreditStyles: false,
  minDepositAmount: 10,
};
