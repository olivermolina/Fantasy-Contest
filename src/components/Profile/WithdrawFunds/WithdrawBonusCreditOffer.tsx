import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';
import { showDollarPrefix } from '~/utils/showDollarPrefix';

interface Props {
  /**
   * Callback function to confirm the offer
   */
  handleSubmit: (cashBalance: number, bonusCredit: number) => void;
  /**
   * Callback function to proceed into the next steps to withdraw cash
   */
  handleNext: () => void;
  /**
   * Boolean to show the offer dialog message
   * @example
   */
  open: boolean;
  /**
   * Callback function to handle the show/close the offer dialog
   */
  setOpen: (open: boolean) => void;
  /**
   * Max bonus credit that the user can get from the offer
   * @example 1000
   */
  maxRetentionBonus: number;
  /**
   * Bonus credit match multiplier
   * @example 2
   */
  retentionBonusMatchMultiplier: number;
  /**
   * User's available withdrawable amount
   * @example 2
   */
  cashBalance: number;
}

/**
 * A withdrawal offer dialog that shows a bonus credit to the user
 * when they keep their money on the platform
 */
export default function WithdrawBonusCreditOffer(props: Props) {
  const {
    handleSubmit,
    handleNext,
    open,
    setOpen,
    cashBalance,
    retentionBonusMatchMultiplier,
    maxRetentionBonus,
  } = props;

  const handleClose = () => {
    setOpen(false);
  };

  const handleCancel = () => {
    handleNext();
    handleClose();
  };

  const matchBonusCredit = cashBalance * retentionBonusMatchMultiplier;
  const finalBonusCredit =
    matchBonusCredit > maxRetentionBonus ? maxRetentionBonus : matchBonusCredit;
  const totalBonusCredit = finalBonusCredit + cashBalance;

  const handleConfirm = () => {
    handleClose();
    handleSubmit(cashBalance, totalBonusCredit);
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        Before you Withdraw Bonus Credit Offer!
      </DialogTitle>
      <DialogContent>
        We will offer you{' '}
        <span className="font-semibold">
          {showDollarPrefix(totalBonusCredit, true)}
        </span>{' '}
        in <span className="font-semibold">BONUS CREDIT</span> if you keep your
        money on the platform to play! By pressing confirm we will convert all
        your Cash Balance{' '}
        <span className="font-semibold">
          ({showDollarPrefix(cashBalance, true)})
        </span>{' '}
        to Bonus Credit and add an additional{' '}
        <span className="font-semibold">
          {showDollarPrefix(finalBonusCredit, true)}
        </span>
        !
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCancel}>Cancel</Button>
        <Button onClick={handleConfirm} variant={'contained'} autoFocus>
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  );
}
