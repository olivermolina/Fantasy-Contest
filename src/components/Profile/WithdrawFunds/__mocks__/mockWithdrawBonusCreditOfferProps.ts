export const mockWithdrawBonusCreditOfferProps = {
  handleSubmit: () => console.log('submit'),
  handleNext: () => console.log('next'),
  open: true,
  setOpen: () => console.log('set open'),
  maxRetentionBonus: 500,
  retentionBonusMatchMultiplier: 2,
  cashBalance: 100,
};
