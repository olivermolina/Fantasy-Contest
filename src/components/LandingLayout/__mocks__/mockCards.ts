import { contestCategoriesMock } from '~/components/Pages/Admin/ManageFreeSquarePromotion/__mocks__/contestCategoriesMock';

export const mockCards = [
  {
    id: '1',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/dallas-goedert.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: {
      id: '1',
      discount: 98,
      pickCategories: contestCategoriesMock,
      maxStake: 25,
      freeEntryEnabled: false,
    },
  },
  {
    id: '2',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/patrick-mahomes.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
  {
    id: '3',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/tyrod-taylor.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
  {
    id: '4',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/tyrod-taylor.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
];
