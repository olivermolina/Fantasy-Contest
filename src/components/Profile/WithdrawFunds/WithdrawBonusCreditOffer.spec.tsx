import { render } from '@testing-library/react';
import WithdrawBonusCreditOffer from './WithdrawBonusCreditOffer';
import { mockWithdrawBonusCreditOfferProps } from './__mocks__/mockWithdrawBonusCreditOfferProps';

describe('WithdrawBonusCreditOffer', () => {
  it('render correctly', () => {
    const { baseElement } = render(
      <WithdrawBonusCreditOffer {...mockWithdrawBonusCreditOfferProps} />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
