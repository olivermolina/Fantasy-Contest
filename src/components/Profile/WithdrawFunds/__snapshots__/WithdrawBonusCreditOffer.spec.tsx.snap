// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithdrawBonusCreditOffer render correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="alert-dialog-description"
        aria-labelledby="alert-dialog-title"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="alert-dialog-title"
        >
          Before you Withdraw Bonus Credit Offer!
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          We will offer you
           
          <span
            class="font-semibold"
          >
            $300.00
          </span>
           
          in 
          <span
            class="font-semibold"
          >
            BONUS CREDIT
          </span>
           if you keep your money on the platform to play! By pressing confirm we will convert all your Cash Balance
           
          <span
            class="font-semibold"
          >
            (
            $100.00
            )
          </span>
           
          to Bonus Credit and add an additional
           
          <span
            class="font-semibold"
          >
            $200.00
          </span>
          !
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-focusVisible MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Confirm
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            >
              <span
                class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
              >
                <span
                  class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
                />
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
